keps/sig-node/20190226-pod-overhead.md

```diff
@@ -93,157 +117,120 @@ so should not be impacted by pod Overhead.
 
 #### Pod overhead
-Introduce a Pod.Spec.Resources field on the pod to specify the pods overhead.
+Introduce a Pod.Spec.Overhead field on the pod to specify the pods overhead.
 
 ```
 Pod {
   Spec PodSpec {
-    // Overhead is the resource overhead consumed by the Pod, not including
-    // container resource usage. Users should leave this field unset.
+    // Overhead is the resource overhead incurred from the runtime.
     // +optional
     Overhead *ResourceRequirements
   }
 }
-```
 
-For scheduling, the pod resource requests are added to the container resource requests.
+For scheduling, the pod `Overhead` resource requests are added to the container resource requests.
 
 We don't currently enforce resource limits on the pod cgroup, but this becomes feasible once
 pod overhead is accountable. If the pod specifies a resource limit, and all containers in the
 pod specify a limit, then the sum of those limits becomes a pod-level limit, enforced through the
 pod cgroup.
 
-Users are not expected to manually set the pod resources; if a runtimeClass is being utilized,
-the manual value will be discarded. See RuntimeController for the proposal for setting these
-resources.
+Users are not expected to manually set `Overhead`; any prior values will being set will result in the workload
+being rejected. If runtimeClass is configured and selected in the PodSpec, `Overhead` will be set to the value
+defined in the corresponding runtimeClass. This is described in detail in
+[RuntimeClass admission controller](#runtimeclass-admission-controller).
+
+Being able to specify resource requirements for a workload at the pod level instead of container
+level has been discussed, but isn't proposed in this KEP.
+
+In the event that pod-level requirements are introduced, pod overhead should be kept separate. This
+simplifies several scenarios:
+ - overhead, once added to the spec, stays with the workload, even if runtimeClass is redefined
+ or removed.
+ - the pod spec can be referenced directly from scheduler, resourceQuote controller and kubelet,
+ instead of referencing a runtimeClass object which could have possibly been removed.
 
-Being able to specify resource requirements for a workload at the pod level instead of container level
-has been discussed, but isn't proposed in this KEP.
+#### Container Runtime Interface (CRI)
 
-In the event that pod-level requirements are introduced, pod overhead should be kept separate. This simplifies
-several scenarios:
- - overhead, once added to the spec, stays with the workload, even if runtimeClass is redefined or removed.
- - the pod spec can be referenced directly from scheduler, resourceQuote controller and kubelet, instead of referencing
- a runtimeClass object which could have possibly been removed.
+The pod cgroup is managed by the Kubelet, so passing the pod-level resource to the CRI implementation
+is not strictly necessary. However, some runtimes may wish to take advantage of this information, for
```

Comment:
Providing overhead across CRI is necessary for Windows as we implement a runtimeclass for Hyper-V. There's no cgroup that a limit can be inherited from, so the sandbox resources need to be determined at creation time. OS overhead will be on the order of hundreds of MB for a Windows sandbox and may vary release to release as things are tuned and features are added or cut.

Reply:
Thanks @PatrickLang - with each release version, then, the runtimeClass should be updated.  Glad to hear this CRI addition is useful for your use case as well.